Makefile-v2.mk

```diff
@@ -0,0 +1,296 @@
+##   agent-image     Builds agent Docker image.
+##   agentctl-image  Builds agentctl Docker image.
+##   operator-image  Builds operator Docker image.
+##   crow-image      Builds crow Docker image.
+##   smoke-image     Builds smoke test Docker image.
+##
+## Targets for packaging:
+##
+##   dist                   Produce release assets for everything.
+##   dist-agent-binaries    Produce release-ready agent binaries.
+##   dist-agentctl-binaries Produce release-ready agentctl binaries.
+##   dist-packages          Produce release-ready DEB and RPM packages.
+##   dist-agent-intaller    Produce a Windows installer for Grafana Agent.
+##
+## Targets for generating assets:
+##
+##   generate             Generate everything.
+##   generate-crds        Generate Grafana Agent Operator CRDs.
+##   generate-manifests   Generate production/kubernetes YAML manifests.
+##   generate-dashboards  Generate dashboards in example/docker-compose after
+##                        changing Jsonnet.
+##   generate-protos      Generate protobuf files.
+##
+## Other targets:
+##
+##   build-container-cache  Create a cache for the build container to speed up
+##                          subsequent proxied builds
+##   clean                  Clean caches and built binaries
+##   help                   Displays this message
+##   info                   Print Makefile-specific environment variables
+##
+## Environment variables:
+##
+##   USE_CONTAINER    Set to 1 to enable proxying commands to build container
+##   AGENT_IMAGE      Image name:tag built by `make agent-image`
+##   AGENTCTL_IMAGE   Image name:tag built by `make agentctl-image`
+##   OPERATOR_IMAGE   Image name:tag built by `make operator-image`
+##   CROW_IMAGE       Image name:tag built by `make crow-image`
+##   SMOKE_IMAGE      Image name:tag built by `make smoke-image`
+##   BUILD_IMAGE      Image name:tag used by USE_CONTAINER=1
+##   AGENT_BINARY     Output path of `make agent` (default build/agent)
+##   AGENTCTL_BINARY  Output path of `make agentctl` (default build/agentctl)
+##   OPERATOR_BINARY  Output path of `make operator` (default build/agent-operator)
+##   CROW_BINARY      Output path of `make crow` (default build/agent-crow)
+##   SMOKE_BINARY     Output path of `make smoke` (default build/agent-smoke)
+##   GOOS             Override OS to build binaries for
+##   GOARCH           Override target architecture to build binaries for
+##   GOARM            Override ARM version (6 or 7) when GOARCH=arm
+##   CGO_ENABLED      Set to 0 to disable Cgo for builds
+##   RELEASE_BUILD    Set to 1 to build release binaries
+##   VERSION          Version to inject into built binaries.
+##   GO_TAGS          Extra tags to use when building.
+
+# TODO(rfratto): test-packages target
+#
+# This depends on some reworking of how the Go tests in ./packaging works to
+# assume that packages have already been built so we don't have to hook in
+# packaging code here.
```

Comment:
I'm leaving this purposefully unfinished for this PR, I'll add the target once this replaces the existing Makefile.